### PR TITLE
Handle a lot of files on Linux

### PR DESCRIPTION
**Pull request by ivolimasilva**
This is a problem found in https://github.com/sindresorhus/trash-cli/issues/9. The problem isn't with [sindresorhus/trash-cli](https://github.com/sindresorhus/trash-cli) but with [sindresorhus/trash](https://github.com/sindresorhus/trash) since it's here where the call to the binary is made.

The solution was to use [sindresorhus/p-map](https://github.com/sindresorhus/p-map) and make single calls with limited concurrency.

This fix is relative to Linux. I tried fixing it on Windows too but I get `Incorrect value of args option` every time I call the binary which also is the reason the tests all crash on my machine (Windows 10 Pro, 64 bits, v1709, build 16299.19). I don't have access to a macOS machine.

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -101,6 +101,20 @@ test('directories', async t => {
...
 });
 
+test('tons of files', async t => {
+	const paths = [];
+	for (let i = 0; i < 10000; i++) {
+		paths.push('file' + i);
+		fs.writeFileSync('file' + i, '');
+	}
+
+	await m(paths);
```
Comment (sindresorhus) [inline test.js]: This needs to use `t.notThrows`

File: test.js [hunk 2]
```diff
@@ -101,6 +101,20 @@ test('directories', async t => {
 	t.false(fs.existsSync(321));
 });
 
+test('tons of files', async t => {
+	const paths = [];
+	for (let i = 0; i < 10000; i++) {
```
Comment (sindresorhus) [inline test.js]: Reduce this to 5000. 10000 is too slow. Also move the number into a constant so it's easily modified, and reused on line 113.

**Comment by sindresorhus** [comment]
The above description is not entirely correct as the linked issue is about macOS and too many CLI arguments, but the tests do reproduce the issue on macOS and this does fix an issue on Linux with too many files.

**Comment by sindresorhus** [comment]
> I tried fixing it on Windows too but I get Incorrect value of args option every time I call the binary which also is the reason the tests all crash on my machine (Windows 10 Pro, 64 bits, v1709, build 16299.19). 

Strange. It works fine on my Windows 10 VM:

```
 tons of files

  C:\Users\SINDRE~1\AppData\Local\Temp\51277be0-7af0-454e-bd8d-140cc2c4f4a4\node_modules\pify\index.js:49

  Rejected promise returned by test. Reason:

  Error {
    code: 'ENAMETOOLONG',
    errno: 'ENAMETOOLONG',
    syscall: 'spawn',
    message: 'spawn ENAMETOOLONG',
  }

  P (node_modules/pify/index.js:49:6)
  ret (node_modules/pify/index.js:72:32)
  module.exports.paths (lib/win.js:9:41)
  module.exports (index.js:23:24)
  Test.__dirname [as fn] (test.js:111:8)
```

Where is the `Incorrect value of args option` error coming from?

**Comment by ivolimasilva** [comment]
Running the test suite on Windows gives me `Incorrect value of args option` on Pify every test.

```
Rejected promise returned by test. Reason:

  TypeError {
    message: 'Incorrect value of args option',
  }

  P (node_modules/pify/index.js:49:6)
  node_modules/pify/index.js:11:9
  ret (node_modules/pify/index.js:72:32)
  path (lib/win.js:11:61)
  Promise.resolve.then.el (node_modules/p-map/index.js:46:16)
```

Calling the Trash API from another script, gives a similiar error:

```
TypeError: Incorrect value of args option
    at exports.execFile (child_process.js:197:11)
    at P (C:\Users\ivoli\Projectos\trash\node_modules\pify\index.js:49:6)
    at Promise (<anonymous>)
    at C:\Users\ivoli\Projectos\trash\node_modules\pify\index.js:11:9
    at ret (C:\Users\ivoli\Projectos\trash\node_modules\pify\index.js:72:32)
    at path (C:\Users\ivoli\Projectos\trash\lib\win.js:11:61)
    at Promise.resolve.then.el (C:\Users\ivoli\Projectos\trash\node_modules\p-map\index.js:46:16)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
    at Function.Module.runMain (module.js:611:11)
```

**Comment by sindresorhus** [comment]
The error is coming from https://github.com/nodejs/node/blob/556ebab30e171c2941deb529a42ad47301972af5/lib/child_process.js#L198 which is really weird. If I were to guess, I would say `npm` is resolving the incorrect dependency versions of some dependencies, which has been a known problem. Make sure you're on the absolute latest version of npm.

File: test.js [hunk 3]
```diff
@@ -102,15 +102,16 @@ test('directories', async t => {
 });
 
 test('tons of files', async t => {
+	const nFiles = 5000;
```
Comment (sindresorhus) [inline test.js]: `nFiles` => `FILE_COUNT`

**Pull request merged**